### DISCUSSION TITLE
unable to verify the first certificate

### DISCUSSION THREAD

**Discussion by simplenotezy** [laravel/valet#1127] [Q&A]
I am running Valet to serve as an API endpoint for a VUE/Nuxt application. However, I am getting this error:

    NuxtServerError: unable to verify the first certificate

Possibly because of valet using a self-signed certificate. What can I do to overcome this?

**Comment by drbyte** [comment]
> Possibly because of valet using a self-signed certificate.

Valet's certificate authority is in your `/Users/YOURUSERNAME/.config/valet/CA` directory. 
To overcome the fact that it's a self-signed certificate, Valet already imports that certificate to your Mac's keychain so that keychain-supported browsers will automatically trust it, and also tells Nginx about it so it knows how to link up with the site-specific certificates in your sites' nginx configs.

I'm not sure if Nuxt supports specifying a certificate authority (ie: "who" signed the site-specific certificate). It appears that you can specify the site-specific certificate details: https://nuxtjs.org/api/configuration-server

You didn't say whether your browser can properly access the endpoint without errors.

I suppose you could explore whether you can add and leverage the more detailed configuration features mentioned here: https://github.com/herbertscruz/nuxt-server including disabling verification ... but that then risks you going into adding code to your app that only benefits your dev environment.

You might try proxying through Valet's nginx by customizing the nginx config for the site, setting the proxy_pass value. The NuxtJs docs give examples of this for Forge, which Valet is very similar to.

Worst case you could just avoid using https at all for this endpoint given you're in a dev-only environment.

Maybe others have found a different workaround. But I'd be inclined to just skip https for the custom server while in dev.

**Comment by simplenotezy** [comment]
Thanks for clarification!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by gilles6** [comment]
@simplenotezy How do you configure **Valet** and **Nuxt** to serve **Nuxt** website ?
When I try, I get a `404 - Not Found` error in my browser.

**Reply by simplenotezy** [reply]
@gilles6 Just now I got a notification about this message, so apologies for late response. Export your Nuxt.js application as a static website (nuxt generate), then go into the `dist` folder and run `valet link`. That should make the website available at `dist.test`. I usually create a symlink though, so I have an url like `my-project.test` that is symlinked to the `dist` folder.

**Comment by jonsherrard** [comment]
Just saving anyone who is here from Google.

You can run your local node application, NUXT, Gatsby, whatever with the following ENV variable

NODE_TLS_REJECT_UNAUTHORIZED=0

Which will disable Node's SSL auth. 

https://nodejs.org/api/cli.html#cli_node_tls_reject_unauthorized_value

For the love of god... only do this in dev.

**Comment by scherii** [comment]
1. Put this line in your `.zshrc` / `.bashrc` / `.profile`:
`export NODE_EXTRA_CA_CERTS="$HOME/.config/valet/CA/LaravelValetCASelfSigned.pem"`
2. Reboot
3. 🎉 `unable to verify the first certificate` is gone

Further information: https://nodejs.org/api/cli.html#node_extra_ca_certsfile

**Reply by eom-eom** [reply]
@scherii this saves my ass thanks!

**Reply by aligajani** [reply]
This solved it thanks

**Comment by AnzhiZhang** [comment]
For me, I use this website: https://myssl.com/chain_download.html to download a full certificate chain, it works.
